File: src/test/java/dev/sanket/twosumsortedarray/SolutionTest.java
package dev.sanket.twosumsortedarray;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import static org.assertj.core.api.Assertions.assertThat;


class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @MethodSource("testData")
    public void shouldPass(int[] numbers, int target, int[] expected) {
        int[] indices = solution.twoSum(numbers, target);

        assertThat(indices).containsExactly(expected);
    }

    private static Stream<Arguments> testData() {
        return Stream.of(
            Arguments.of(new int[]{2, 7, 11, 15}, 9, new int[]{1, 2}),
            Arguments.of(new int[]{2, 3, 4}, 6, new int[]{1, 3}),
            Arguments.of(new int[]{-1, 0}, -1, new int[]{1, 2}),
            Arguments.of(new int[]{5, 25, 75}, 100, new int[]{2, 3}),
            Arguments.of(new int[]{-10, -8, -2, 1, 2, 5, 6}, 0, new int[]{3, 5})
        );
    }
}

File: src/test/kotlin/dev/sanket/rotatearray/SolutionTest.kt
package dev.sanket.rotatearray

import assertk.assertThat
import assertk.assertions.containsExactly
import dev.sanket.dev.sanket.rotatearray.Solution
import java.util.stream.Stream
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class SolutionTest {

    private val solution = Solution()

    @ParameterizedTest
    @MethodSource("testData")
    fun shouldPass(nums: IntArray, k: Int, expected: IntArray) {
        solution.rotate(nums, k)
        assertThat(nums).containsExactly(*expected)
    }

    companion object {

        @JvmStatic
        fun testData(): Stream<Arguments> = Stream.of(
            Arguments.of(intArrayOf(1, 2, 3, 4, 5, 6, 7), 0, intArrayOf(1, 2, 3, 4, 5, 6, 7)),
            Arguments.of(intArrayOf(1, 2, 3, 4, 5, 6, 7), 7, intArrayOf(1, 2, 3, 4, 5, 6, 7)),
            Arguments.of(intArrayOf(1, 2, 3, 4, 5, 6, 7), 3, intArrayOf(5, 6, 7, 1, 2, 3, 4)),
            Arguments.of(intArrayOf(-1, -100, 3, 99), 2, intArrayOf(3, 99, -1, -100)),
            Arguments.of(
                intArrayOf(2147483647, -2147483648, 33, 219, 0),
                4,
                intArrayOf(-2147483648, 33, 219, 0, 2147483647)
            ),
        )
    }
}